src/components/Pages/Homepage.js

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import NavBar from '../Navbar/Navbar';
+import FishCard from '../Fishcard/Fishcard';
+import './HomePage.css';
+import { getFish } from '../../redux/Fish/Fish';
+
+const Homepage = () => {
+  const navigate = useNavigate();
+  const fishArray = useSelector((state) => state.fish.fishArray);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!fishArray.length) dispatch(getFish());
+  }, []);
+
+  return (
+    <>
+      <header className="header">
+        <NavBar title="PerspectiveWild" />
+        <img className="home-img" src="./images/kindpng_7177875.png" alt="im" />
+        <div className="image-cover" />
+      </header>
+      <h6>Animal species</h6>
+      <main className="main">
+        {fishArray.map((fish) => (
+          <FishCard
+            key={fish.id}
+            name={fish.name}
+            img={fish.img}
+            weight={fish.maxWeight}
+            pos={fish.pos}
+            onClick={() => { navigate('/detailsPage', { state: { fish } }); }}
+          />
```

Comment:
- [ ] Kindly add functionality for this project requirement `The Home page has a list of categories that could be filtered using given parameters, e.g. category name.` .This can be achieved by either adding a search input field for users to search a category name or other ways you would think about.

![search](https://user-images.githubusercontent.com/53124548/164763491-717a9a50-89a9-4435-bea7-dc24031e89ec.png)

Reply:
thank you for pointing that out. I'll take care of it